lib/cloudregister/registerutils.py

```diff
@@ -516,6 +517,59 @@ def get_credentials(credentials_file):
     return (username, password)
 
 
+# ----------------------------------------------------------------------------
+def setup_registry(registry_fqdn, username, password):
+    """Set all the necessary parts for the registry,
+       returns True if the setup completed, False otherwise."""
+    if not os.path.exists(os.path.dirname(REGISTRY_CREDENTIALS_PATH)):
+        os.makedirs(os.path.dirname(REGISTRY_CREDENTIALS_PATH))
+
+    setup_registry_succeed = set_registry_auth_token(
+        registry_fqdn, username, password
+    )
+    return setup_registry_succeed
+
+
+# ----------------------------------------------------------------------------
+def set_registry_auth_token(registry_fqdn, username, password):
+    """Set the auth token to pull images from SUSE registry."""
+    auth_token = base64.b64encode('{username}:{password}'.format(
+        username=username,
+        password=password
+    ).encode()).decode()
+    registry_credentials = {registry_fqdn: {'auth': auth_token}}
+    config_json = {}
+    if not os.path.exists(REGISTRY_CREDENTIALS_PATH):
+        config_json.update({'auths': registry_credentials})
+    else:
+        try:
+            with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
+                config_json = json.load(cred_json)
+            # set the new registry credentials,
+            # independently of what that content was
+            config_json['auths'].update(registry_credentials)
+        except KeyError:
+            # file does not exist or "auths" key is not set
```

Comment:
When we get here we know the file exists, we already checked that condition. So the comment is incorrect. Also why not use a way that avoids exceptions when we can. Exceptions are for exceptional cases that are hard to test or that we truly cannot expect.

We can do positive testing and avoid some exceptions as follows: (Also handles the case to preserve the file we couldn't parse, rather than clobbering it)

```
if os.path.exists(REGISTRY_CREDENTIALS_PATH):
    try:
        with open(REGISTRY_CREDENTIALS_PATH, 'r') as cred_json:
            container_engine_config = json.load(cred_json)
        if 'auths' in container_engine_config:
            container_engine_config['auths'].update(registry_credentials)
    except Error:
        logging.info('Unable to parse existing %s, preserving file as %s.bak, writing new credentials') % (REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH)
        os.system('mv -Z %s %s.bak' % (REGISTRY_CREDENTIALS_PATH, REGISTRY_CREDENTIALS_PATH)
else:
    container_engine_config = {'auths': registry_credentials}
```

Reply:
I appreciate the suggestion, wasn't trying to be difficult with the coding styles

I don't have a strong preference for any of the two styles, followed the one preferred by the community (EAFP) even if it felt more natural the other one (coming from other languages like C/C++) 
I can see benefits/pros in both and I can see cons in both as well

I guess or understand you have a preference for one of them, I will follow that style then (as I said, no strong preference), for files/dictionaries (like checking the key in this case but especially with files) I find the `try-except` approach handy, easier to read and to code than the suggested code. Still will use the suggested style as the main one and keep the `try-except` as the less common approach

I understand the suggestion is pseudo code and I will take the preservation of the file  into account

Reply:
any reason about why using `mv` instead of `cp` ?